tests/devlxd-container

```diff
@@ -83,7 +83,12 @@ if hasNeededAPIExtension instance_ready_state; then
   [ "$(lxc config get c1 volatile.last_state.ready)" = "false" ]
 
   lxc start c1
-  waitInstanceBooted c1
+
+  # Wait for the instance to report some processes, don't wait for systemctl as it can sometimes boot into a "degraded" state after a force stop.
```

Comment:
`waitInstanceBooted()` is supposed to carry on even if `systemctl is-system-running --wait` reports a degraded state. In such case it should just log a warning but still return 0. Is that not what you observed?

Looking at https://github.com/canonical/lxd-ci/actions/runs/10503168232/job/29096039000#step:8:483, it seems that it hit the timeout of 90s and couldn't get an IPv4.

Could the lack of IPv4 explain why DNS resolution failed in your PR test (https://github.com/canonical/lxd-ci/actions/runs/10510684542/job/29119965810#step:8:489) ?

Reply:
Looking at your PR test, after the DNS failure, it shows that indeed `c1` has no IPv4. I'd be curious to see `lxc exec c1 -- systemctl --failed` prior to the `systemctl start snapd` you added. Maybe there's a service we can disable to make it more reliable.

Reply:
Ahh yes my bad. I'll have to get another tmate session running on failure.

Reply:
This is the same unit that failed in the last tmate session I had:
```
runner@fv-az1755-453:~/work/lxd-ci/lxd-ci$ sudo su
root@fv-az1755-453:/home/runner/work/lxd-ci/lxd-ci# lxc list
+------+---------+------+-----------------------------------------------+-----------+-----------+
| NAME |  STATE  | IPV4 |                     IPV6                      |   TYPE    | SNAPSHOTS |
+------+---------+------+-----------------------------------------------+-----------+-----------+
| c1   | RUNNING |      | fd42:4e31:ad29:4b70:216:3eff:fe36:e4e7 (eth0) | CONTAINER | 0         |
+------+---------+------+-----------------------------------------------+-----------+-----------+
root@fv-az1755-453:/home/runner/work/lxd-ci/lxd-ci# lxc exec c1 -- systemctl --failed
  UNIT                                 LOAD   ACTIVE SUB    DESCRIPTION                      
● systemd-networkd-wait-online.service loaded failed failed Wait for Network to be Configured

Legend: LOAD   → Reflects whether the unit definition was properly loaded.
        ACTIVE → The high-level unit activation state, i.e. generalization of SUB.
        SUB    → The low-level unit activation state, values depend on unit type.

1 loaded units listed.
root@fv-az1755-453:/home/runner/work/lxd-ci/lxd-ci# 
```

We certainly need this unit to be running. Otherwise we won't be able to install LXD. I'm not sure why this unit is failing to come up after the force stop. `journalctl` isn't showing anything useful:
```
root@c1:~# journalctl -u systemd-networkd-wait-online
Aug 23 08:36:08 c1 systemd[1]: Starting systemd-networkd-wait-online.service - Wait for Network to be Configured...
-- Boot 06cea08df49a4e46b5da440d42e24b91 --
Aug 23 08:36:17 c1 systemd[1]: Starting systemd-networkd-wait-online.service - Wait for Network to be Configured...
Aug 23 08:38:18 c1 systemd-networkd-wait-online[244]: Timeout occurred while waiting for network connectivity.
Aug 23 08:38:18 c1 systemd[1]: systemd-networkd-wait-online.service: Main process exited, code=exited, status=1/FAILURE
Aug 23 08:38:18 c1 systemd[1]: systemd-networkd-wait-online.service: Failed with result 'exit-code'.
Aug 23 08:38:18 c1 systemd[1]: Failed to start systemd-networkd-wait-online.service - Wait for Network to be Configured.
Aug 23 08:41:16 c1 systemd[1]: Starting systemd-networkd-wait-online.service - Wait for Network to be Configured...
Aug 23 08:43:16 c1 systemd-networkd-wait-online[412]: Timeout occurred while waiting for network connectivity.
Aug 23 08:43:16 c1 systemd[1]: systemd-networkd-wait-online.service: Main process exited, code=exited, status=1/FAILURE
Aug 23 08:43:16 c1 systemd[1]: systemd-networkd-wait-online.service: Failed with result 'exit-code'.
Aug 23 08:43:16 c1 systemd[1]: Failed to start systemd-networkd-wait-online.service - Wait for Network to be Configured.
Aug 23 08:43:53 c1 systemd[1]: Starting systemd-networkd-wait-online.service - Wait for Network to be Configured...
Aug 23 08:45:53 c1 systemd-networkd-wait-online[437]: Timeout occurred while waiting for network connectivity.
Aug 23 08:45:53 c1 systemd[1]: systemd-networkd-wait-online.service: Main process exited, code=exited, status=1/FAILURE
Aug 23 08:45:53 c1 systemd[1]: systemd-networkd-wait-online.service: Failed with result 'exit-code'.
Aug 23 08:45:53 c1 systemd[1]: Failed to start systemd-networkd-wait-online.service - Wait for Network to be Configured.
```

Reply:
the container doesn't have an IP, suggesting the network service itself hasn't started/finished.

Reply:
I've got this:
```
Aug 23 09:23:30 c1 systemd-networkd[244]: Failed to increase receive buffer size for general netlink socket, ignoring: Operation not permitted
```
Otherwise there an awful lot of errors from `udevadm. They're all permission denied errors, e.g.:
```
Aug 23 09:23:30 c1 udevadm[68]: 8021q: Failed to write 'add' to '/sys/module/8021q/uevent': Permission denied
Aug 23 09:23:30 c1 udevadm[68]: 8250: Failed to write 'add' to '/sys/module/8250/uevent': Permission denied
Aug 23 09:23:30 c1 udevadm[68]: acpi: Failed to write 'add' to '/sys/module/acpi/uevent': Permission denied
Aug 23 09:23:30 c1 udevadm[68]: acpi_cpufreq: Failed to write 'add' to '/sys/module/acpi_cpufreq/uevent': Permission denied
Aug 23 09:23:30 c1 udevadm[68]: acpiphp: Failed to write 'add' to '/sys/module/acpiphp/uevent': Permission denied
Aug 23 09:23:30 c1 udevadm[68]: aesni_intel: Failed to write 'add' to '/sys/module/aesni_intel/uevent': Permission denied
Aug 23 09:23:30 c1 udevadm[68]: apparmor: Failed to write 'add' to '/sys/module/apparmor/uevent': Permission denied
Aug 23 09:23:30 c1 udevadm[68]: ata_generic: Failed to write 'add' to '/sys/module/ata_generic/uevent': Permission denied
Aug 23 09:23:30 c1 udevadm[68]: ata_piix: Failed to write 'add' to '/sys/module/ata_piix/uevent': Permission denied
Aug 23 09:23:30 c1 udevadm[68]: autofs4: Failed to write 'add' to '/sys/module/autofs4/uevent': Permission denied
Aug 23 09:23:30 c1 udevadm[68]: binfmt_misc: Failed to write 'add' to '/sys/module/binfmt_misc/uevent': Permission denied
Aug 23 09:23:30 c1 udevadm[68]: blk_cgroup: Failed to write 'add' to '/sys/module/blk_cgroup/uevent': Permission denied
Aug 23 09:23:30 c1 udevadm[68]: blk_crypto: Failed to write 'add' to '/sys/module/blk_crypto/uevent': Permission denied
Aug 23 09:23:30 c1 udevadm[68]: block: Failed to write 'add' to '/sys/module/block/uevent': Permission denied
Aug 23 09:23:30 c1 udevadm[68]: br_netfilter: Failed to write 'add' to '/sys/module/br_netfilter/uevent': Permission denied
Aug 23 09:23:30 c1 udevadm[68]: bridge: Failed to write 'add' to '/sys/module/bridge/uevent': Permission denied
Aug 23 09:23:30 c1 udevadm[68]: button: Failed to write 'add' to '/sys/module/button/uevent': Permission denied
Aug 23 09:23:30 c1 udevadm[68]: ccp: Failed to write 'add' to '/sys/module/ccp/uevent': Permission denied
```

Reply:
Indeed, `dhclient` is no longer part of the default install. The `Failed to increase receive buffer size for general netlink socket` error is something I get on most of my containers but it doesn't seem to impact their connectivity. The `udevadm` errors are to be expected in containers AFAICT.